go/ql/lib/semmle/go/security/Xss.qll

```diff
@@ -127,4 +127,20 @@ module SharedXss {
       )
     }
   }
+
+  /**
+   * A `Template` from `html/template` will HTML-escape data automatically
+   * and therefore acts as a sanitizer for XSS vulnerabilities.
+   */
+  class HtmlTemplateSanitizer extends Sanitizer, DataFlow::Node {
+    HtmlTemplateSanitizer() {
+      exists(Method m, DataFlow::CallNode call | m = call.getCall().getTarget() |
```

Comment:
As a random side-note, I've realised recently that we use `getTarget()` way too much. It only works for direct function calls, not calls to a variable which has function type. `getACalleeIncludingExternals()` deals with that, as well as virtual calls. No need to change it here, but it's something to be aware of.

Reply:
That's good to know, thanks! I had mainly just based this `Sanitizer` on the others in this file which used that pattern.